diff --git a/dist/index.js b/dist/index.js
index cfde8913b2bdaeb99674a7a63706cd84b5ff5ac8..f27634db7b445595a0c55292ff9354a503b81145 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -1032,20 +1032,28 @@ function applyPutPatch(doc, path, patch) {
 }
 function applyInsertPatch(doc, path, patch) {
     let { obj: parent, prop } = pathElemAt(path, -1);
-    if (!Array.isArray(parent)) {
-        throw new RangeError(`target is not an array for patch`);
+    if (Array.isArray(parent)) {
+        if (!(typeof prop === "number")) {
+            throw new RangeError(`index is not a number for patch`);
+        }
+        parent.splice(prop, 0, ...patch.values);
     }
-    if (!(typeof prop === "number")) {
-        throw new RangeError(`index is not a number for patch`);
+    else if (typeof parent === "object" && parent !== null) {
+        parent[prop] = patch.values[0];
+    }
+    else {
+        throw new RangeError(`target is not an array or object for patch`);
     }
-    parent.splice(prop, 0, ...patch.values);
 }
 function applyDelPatch(doc, path, patch) {
     let { obj: parent, prop, parentPath } = pathElemAt(path, -1);
-    if (!(typeof prop === "number")) {
+    if (typeof prop === "string" && typeof parent === "object" && parent !== null && !Array.isArray(parent)) {
+        delete parent[prop];
+    }
+    else if (!(typeof prop === "number")) {
         throw new RangeError(`index is not a number for patch`);
     }
-    if (Array.isArray(parent)) {
+    else if (Array.isArray(parent)) {
         parent.splice(prop, patch.length || 1);
     }
     else if (typeof parent === "string") {
